src/redis-cli.c

```diff
@@ -1398,8 +1399,10 @@ static void repl(void) {
     while((line = linenoise(context ? config.prompt : "not connected> ")) != NULL) {
         if (line[0] != '\0') {
             argv = cliSplitArgs(line,&argc);
-            if (history) linenoiseHistoryAdd(line);
-            if (historyfile) linenoiseHistorySave(historyfile);
+            if (strcasecmp(argv[0], "auth")) {
```

Comment:
This check misses the edge, but still valid, case of the user actually issuing something like `1 AUTH foobared`. Put differently, IMO you need to handle the `repeat` logic as well.

Reply:
Confirmed and fixed :)